zilliqa/src/node.rs

```diff
@@ -131,6 +131,26 @@ pub struct Node {
     message_sender: MessageSender,
     reset_timeout: UnboundedSender<Duration>,
     pub consensus: Consensus,
+    pub chain_id: ChainId,
+}
+
+#[derive(Debug)]
+pub struct ChainId {
+    pub eth: u64,
+    pub zil: u64,
+}
+
+impl ChainId {
+    pub fn new(eth_chain_id: u64) -> Self {
+        ChainId {
+            eth: eth_chain_id,
+            zil: Self::zil_chain_id(eth_chain_id),
+        }
+    }
+
+    pub fn zil_chain_id(eth_chain_id: u64) -> u64 {
+        eth_chain_id - 0x8000
+    }
 }
```

Comment:
Since we're always setting the `zil` chain ID to `eth - 0x8000` can we instead do
```suggestion
#[derive(Debug)]
pub struct ChainId {
    pub eth: u64,
}

impl ChainId {
    pub fn new(eth_chain_id: u64) -> Self {
        ChainId {
            eth: eth_chain_id,
        }
    }

    pub fn zil(&self) -> u64 {
        self.eth - 0x8000
    }
}
```
?